app/src/main/java/com/sopt/now/ui/signup/SignUpActivity.kt

```diff
@@ -2,56 +2,75 @@ package com.sopt.now.ui.signup
 import com.sopt.now.data.Key.USER_PROFILE
+import com.sopt.now.data.ServicePool.authService
 import com.sopt.now.data.User
+import com.sopt.now.data.dto.request.RequestSignUpDto
+import com.sopt.now.data.dto.response.ResponseSignUpDto
 import com.sopt.now.databinding.ActivitySignUpBinding
 import com.sopt.now.ui.login.LoginActivity
 
 class SignUpActivity : AppCompatActivity() {
     private lateinit var binding: ActivitySignUpBinding
-    private lateinit var user: User
+    private lateinit var id: String
+    private lateinit var password: String
+    private lateinit var nickname: String
+    private lateinit var phone: String
+
+    private val viewModel by viewModels<SignUpViewModel>()
+
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivitySignUpBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
-        initSignupBtnClickListener()
+        initView()
+        initObserver()
     }
 
-    private fun initSignupBtnClickListener() {
+    private fun initView() {
         binding.btnSignupToSignup.setOnClickListener {
-            val userID = binding.etSignupId.text.toString()
-            val userPw = binding.etSignupPw.text.toString()
-            val userNickname = binding.etSignupNickname.text.toString()
-            val userPhone = binding.etSignupPhone.text.toString()
-
-            user = User(userID, userPw, userNickname, userPhone)
-
-            checkInput(user)
+            viewModel.signUp(getSignUpRequestDto())
         }
     }
 
-    private fun checkInput(user: User) {
-        return when {
-            user.id.length !in 6..10 -> showMessage(getString(R.string.msg_id_fail))
-            user.password.length !in 8..12 -> showMessage(getString(R.string.msg_password_fail))
-            user.name.isNullOrBlank() -> showMessage(getString(R.string.msg_nickname_fail))
-            user.phone.isNullOrBlank() -> showMessage(getString(R.string.msg_phone_fail))
-            else -> checkedSignup(user)
+    private fun initObserver() {
+        viewModel.liveData.observe(this) { state ->
+            Toast.makeText(
+                this@SignUpActivity,
+                state.message,
+                Toast.LENGTH_SHORT,
+            ).show()
+
+            if (state.isSuccess)
+                goToLogin()
         }
     }
 
-    private fun checkedSignup(user: User) {
-        val intent = Intent(this, LoginActivity::class.java)
-        intent.putExtra(USER_PROFILE, user)
+    private fun getSignUpRequestDto(): RequestSignUpDto {
+        id = binding.etSignupId.text.toString()
+        password = binding.etSignupPw.text.toString()
+        nickname = binding.etSignupNickname.text.toString()
+        phone = binding.etSignupPhone.text.toString()
+
+        return RequestSignUpDto(
+            authenticationId = id,
+            password = password,
+            nickname = nickname,
+            phone = phone
+        )
+    }
 
```

Comment:
위에서도 같은 리뷰가 달려있기는 한 것 같은데 id, password, nickname, phone이 여기 함수에서만 사용되고 있는 것 같아 클래스에 선언된 변수는 지우고 아래처럼 해도 될 것 같아요.

```suggestion
    private fun getSignUpRequestDto(): RequestSignUpDto {
        return RequestSignUpDto(
            authenticationId = binding.etSignupId.text.toString(),
            password = binding.etSignupPw.text.toString(),
            nickname = binding.etSignupNickname.text.toString(),
            phone = binding.etSignupPhone.text.toString()
        )
    }

```

Reply:
아하 그렇군요! 수정하겠습니다! 😄